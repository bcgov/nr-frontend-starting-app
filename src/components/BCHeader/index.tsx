import React from 'react';

import {
  Header,
  HeaderName,
  HeaderGlobalBar,
  HeaderGlobalAction,
  Theme
} from '@carbon/react';
import {
  Search,
  Notification,
  Switcher
} from '@carbon/icons-react';

const BCHeader = () => (
  <Theme theme="g100">
    <Header aria-label="BC Gov's NR Sample App" data-testid="header">
      <HeaderName href="/" prefix="BC Gov's" data-testid="header-name">
        NR Sample App
      </HeaderName>
      <HeaderGlobalBar>
        <HeaderGlobalAction aria-label="Search" data-testid="header-button__search">
          <Search size={20} />
        </HeaderGlobalAction>
        <HeaderGlobalAction aria-label="Notifications" data-testid="header-button__notifications">
          <Notification size={20} />
        </HeaderGlobalAction>
        <HeaderGlobalAction aria-label="App Switcher" tooltipAlignment="end" data-testid="header-button__switcher">
          <Switcher size={20} />
        </HeaderGlobalAction>
      </HeaderGlobalBar>
    </Header>
  </Theme>
);

export default BCHeader;
