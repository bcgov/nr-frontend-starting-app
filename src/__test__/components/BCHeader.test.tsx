/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import BCHeader from '../../components/BCHeader';

describe('the Header component', () => {
  it('should have the correct title', () => {
    const { getByTestId } = render(<BCHeader />);

    // Carbon Header component uses &nbsp; to break the space between the prefix
    // (BC Gov's) and the title (NR Sample App), so it's necessary to use \xA0
    // to validate correctly
    expect(getByTestId('header-name').textContent).toBe('BC Gov\'s\xA0NR Sample App');
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(<BCHeader />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
