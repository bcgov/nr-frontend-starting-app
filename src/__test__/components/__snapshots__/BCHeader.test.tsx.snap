// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Header component should match the snapshot 1`] = `
<div
  className="cds--g100 cds--layer-one"
>
  <header
    aria-label="BC Gov's NR Sample App"
    className="cds--header"
    data-testid="header"
  >
    <a
      className="cds--header__name"
      data-testid="header-name"
      href="/"
    >
      <span
        className="cds--header__name--prefix"
      >
        BC Gov's
      </span>
      Â 
      NR Sample App
    </a>
    <div
      className="cds--header__global"
    >
      <span
        className="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--bottom cds--tooltip cds--icon-tooltip"
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          aria-describedby={null}
          aria-label="Search"
          aria-labelledby="tooltip-11"
          aria-pressed={null}
          className="cds--btn--icon-only cds--header__action cds--btn cds--btn--primary cds--btn--icon-only cds--btn cds--btn--primary"
          data-testid="header-button__search"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={20}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M29,27.5859l-7.5521-7.5521a11.0177,11.0177,0,1,0-1.4141,1.4141L27.5859,29ZM4,13a9,9,0,1,1,9,9A9.01,9.01,0,0,1,4,13Z"
            />
          </svg>
        </button>
        <span
          aria-hidden="true"
          className="cds--popover"
          id="tooltip-11"
          role="tooltip"
        >
          <span
            className="cds--popover-content cds--tooltip-content"
          >
            Search
          </span>
          <span
            className="cds--popover-caret"
          />
        </span>
      </span>
      <span
        className="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--bottom cds--tooltip cds--icon-tooltip"
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          aria-describedby={null}
          aria-label="Notifications"
          aria-labelledby="tooltip-14"
          aria-pressed={null}
          className="cds--btn--icon-only cds--header__action cds--btn cds--btn--primary cds--btn--icon-only cds--btn cds--btn--primary"
          data-testid="header-button__notifications"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={20}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M28.7071,19.293,26,16.5859V13a10.0136,10.0136,0,0,0-9-9.9492V1H15V3.0508A10.0136,10.0136,0,0,0,6,13v3.5859L3.2929,19.293A1,1,0,0,0,3,20v3a1,1,0,0,0,1,1h7v.7768a5.152,5.152,0,0,0,4.5,5.1987A5.0057,5.0057,0,0,0,21,25V24h7a1,1,0,0,0,1-1V20A1,1,0,0,0,28.7071,19.293ZM19,25a3,3,0,0,1-6,0V24h6Zm8-3H5V20.4141L7.707,17.707A1,1,0,0,0,8,17V13a8,8,0,0,1,16,0v4a1,1,0,0,0,.293.707L27,20.4141Z"
            />
          </svg>
        </button>
        <span
          aria-hidden="true"
          className="cds--popover"
          id="tooltip-14"
          role="tooltip"
        >
          <span
            className="cds--popover-content cds--tooltip-content"
          >
            Notifications
          </span>
          <span
            className="cds--popover-caret"
          />
        </span>
      </span>
      <span
        className="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--bottom-right cds--tooltip cds--icon-tooltip"
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          aria-describedby={null}
          aria-label="App Switcher"
          aria-labelledby="tooltip-17"
          aria-pressed={null}
          className="cds--btn--icon-only cds--header__action cds--btn cds--btn--primary cds--btn--icon-only cds--btn cds--btn--primary"
          data-testid="header-button__switcher"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={20}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 4H18V8H14zM4 4H8V8H4zM24 4H28V8H24zM14 14H18V18H14zM4 14H8V18H4zM24 14H28V18H24zM14 24H18V28H14zM4 24H8V28H4zM24 24H28V28H24z"
            />
          </svg>
        </button>
        <span
          aria-hidden="true"
          className="cds--popover"
          id="tooltip-17"
          role="tooltip"
        >
          <span
            className="cds--popover-content cds--tooltip-content"
          >
            App Switcher
          </span>
          <span
            className="cds--popover-caret"
          />
        </span>
      </span>
    </div>
  </header>
</div>
`;
